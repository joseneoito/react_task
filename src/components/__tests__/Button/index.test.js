import {render, cleanup, screen } from '@testing-library/react'
import {BrowserRouter as Router} from 'react-router-dom';
import ButtonComponent from '../../Button'

describe('tests for button', ()=>{
test('should render Button component',()=>{
    render(<ButtonComponent/>)
    const buttonElement = screen.getByTestId('button')
    expect(buttonElement).toBeInTheDocument()
})


it("snapshot button component", () => {
    const mockText = "save";
    const { asFragment } = render(<ButtonComponent text={mockText}/>);
    expect(asFragment(<ButtonComponent text={mockText} />)).toMatchSnapshot();
  });

  it("should render button component with the right text", () => {
    const mockText = "This is just for the sake of the test";
    const { getByText } = render(<ButtonComponent text={mockText} />);
    expect(getByText(mockText)).not.toBeNull();
  });

  it("alternative way using toBeInTheDocument jest-dom uitility library", () => {
    const mockText = "This is just for the sake of the test";
    const { getByText } = render(<ButtonComponent text={mockText} />);
    expect(getByText(mockText)).toBeInTheDocument();
  });

})
