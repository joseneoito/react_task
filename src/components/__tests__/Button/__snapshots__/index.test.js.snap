// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`tests for button snapshot button component 1`] = `
<DocumentFragment>
  <button
    class="MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButtonBase-root  css-sghohy-MuiButtonBase-root-MuiButton-root"
    data-testid="button"
    tabindex="0"
    type="submit"
  >
    save
    <span
      class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
    />
  </button>
</DocumentFragment>
`;
